src/StreamrClient.ts

```diff
@@ -0,0 +1,583 @@
+import EventEmitter from 'eventemitter3'
+// @ts-expect-error
+import { ControlLayer } from 'streamr-client-protocol'
+import Debug from 'debug'
+
+import { counterId, uuid, CacheAsyncFn } from './utils'
+import { validateOptions } from './stream/utils'
+import Config from './Config'
+import StreamrEthereum from './Ethereum'
+import Session from './Session'
+import Connection from './Connection'
+import Publisher from './publish'
+import Subscriber from './subscribe'
+import { getUserId } from './user'
+import { Todo } from './types'
+import { StreamEndpoints, StreamListQuery } from './rest/StreamEndpoints'
+import { LoginEndpoints } from './rest/LoginEndpoints'
+import { DataUnionEndpoints, DataUnionOptions } from './rest/DataUnionEndpoints'
+import { BigNumber } from '@ethersproject/bignumber'
+import Stream, { StreamProperties } from './stream'
+import { ExternalProvider, JsonRpcFetchFunc } from '@ethersproject/providers'
+
+export interface StreamrClientOptions {
```

Comment:
This should go into `./Config.ts` and could probably be largely inferred from defaults.